src/gltf/Raw2Gltf.cpp

```diff
@@ -261,26 +261,31 @@ ModelData* Raw2Gltf(
            * Other values translate directly.
            */
           RawMetRoughMatProps* props = (RawMetRoughMatProps*)material.info.get();
-          // merge metallic into the blue channel and roughness into the green, of a new combinatory
-          // texture
-          aoMetRoughTex = textureBuilder.combine(
-              {
-                  material.textures[RAW_TEXTURE_USAGE_OCCLUSION],
-                  material.textures[RAW_TEXTURE_USAGE_METALLIC],
-                  material.textures[RAW_TEXTURE_USAGE_ROUGHNESS],
-              },
-              "ao_met_rough",
-              [&](const std::vector<const TextureBuilder::pixel*> pixels) -> TextureBuilder::pixel {
-                const float occlusion = (*pixels[0])[0];
-                const float metallic = (*pixels[1])[0];
-                const float roughness = (*pixels[2])[0];
-                return {{occlusion,
-                         props->invertRoughnessMap ? 1.0f - roughness : roughness,
-                         metallic,
-                         1}};
-              },
-              false);
-
+          
+          // If either a metallic or roughness map has been provided, generate the aoMetRoughTex.
+          bool hasMetallicMap = material.textures[RAW_TEXTURE_USAGE_METALLIC] >= 0;
+          bool hasRoughnessMap = material.textures[RAW_TEXTURE_USAGE_ROUGHNESS] >= 0;
+          if (hasMetallicMap || hasRoughnessMap) {
```

Comment:
This should have a hasOcclusionMap check as well, right? As unlikely as it is that a model has an AO map without either metallic or roughness maps, it seems like the right thing to do.

Reply:
I'm not a 3d modeler myself, but I know that we this is a common situation for us. We have a few thousand models that use AO maps but only constants for metalness and roughness. In the tests I've done there haven't been any issues supplying only an AO map and not generating metal/roughness map.

Reply:
Hmm. Confusing. It works with such a combination, even with this change? I don’t set how occlusionTexture would get set. But in any case I realise now, if there’s an AO map but constant roughness & metal, of course we shouldn’t run the texture rebuilder anyway, but rather just copy the source texture. I’ll merge this in a few hours, apologies for the delay!

Reply:
Yeah I think the final gltf was actually just using the occlusion map separately anyway, even when it generated the metalness roughness map. I wondered if that wasn't intentional.